### PR TITLE
Add unit test for residual related functions

### PR DESCRIPTION
**Pull request by Cidana-Developers**
1. add unit tests for ResidualKernel_avx2 ,residual_kernel16bit_sse2_intrin,
   residual_kernel_sub_sampled{w}x{h}_sse_intrin and
   sum_residual8bit_avx2_intrin.

### CODE REVIEW

File: Source/Lib/Common/Codec/EbPictureOperators.c [hunk 1]
```diff
@@ -224,6 +224,60 @@ void residual_kernel_c(
     return;
 }
 
+void residual_kernel_subsampled(
+    uint8_t   *input,
+    uint32_t   inputStride,
```
Comment (andrey-norkin) [inline Source/Lib/Common/Codec/EbPictureOperators.c]: The app and library code uses snake_case for variables names

File: Source/Lib/Common/Codec/EbPictureOperators.h [hunk 2]
```diff
@@ -378,6 +378,17 @@ extern "C" {
         uint32_t  area_width,
         uint32_t  area_height);
 
+    void residual_kernel_subsampled(
+        uint8_t   *input,
```
Comment (andrey-norkin) [inline Source/Lib/Common/Codec/EbPictureOperators.h]: snake_case for variables names

Comment (Cidana-Developers) [reply Source/Lib/Common/Codec/EbPictureOperators.h]: Done.

File: Source/Lib/Common/Codec/EbPictureOperators.c (hunk 1 above)
Comment (Cidana-Developers) [reply Source/Lib/Common/Codec/EbPictureOperators.c]: Done.

**Pull request merged**